src/ring_theory/graded_algebra/radical.lean

```diff
@@ -0,0 +1,332 @@
+/-
+Copyright (c) 2022 Jujian Zhang. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jujian Zhang
+-/
+
+import ring_theory.graded_algebra.homogeneous_ideal
+
+/-!
+# Radical
+
+This file contains a proof that the radical of any homogeneous ideal is a homogeneous ideal
+
+## Main statements
+
+* `homogeneous_ideal.is_prime_iff`: a homogeneous ideal `I` is prime if and only if `I` is
+  homogeneously prime, i.e. if `x, y` are homogeneous elements such that `x * y ∈ I`, then
+  at least one of `x,y` is in `I`.
+* `homogeneous_ideal.rad`: radical of homogeneous ideal is a homogeneous ideal.
+
+## Implementation details
+
+Through out this file, the indexing type `ι` of grading is assumed to be a
+`linear_ordered_cancel_add_comm_monoid`. This might be stronger than necessary and `linearith`
+does not work on `linear_ordered_cancel_add_comm_monoid`.
+
+## Tags
+
+homogeneous, radical
+-/
+
+
+open_locale big_operators
+
+section linear_ordered_cancel_add_comm_monoid
+
+variables {ι : Type*} [linear_ordered_cancel_add_comm_monoid ι] [decidable_eq ι]
+variables {R : Type*} [comm_ring R]
+variables (A : ι → ideal R) [graded_algebra A]
+variable [Π (I : homogeneous_ideal A) (x : R),
+  decidable_pred (λ (i : ι), graded_algebra.proj A i x ∉ I)]
+variable [Π (i : ι) (x : A i), decidable (x ≠ 0)]
+
+lemma homogeneous_ideal.is_prime_iff
```

Comment:
This isn't an `iff` - was it intended to be?

Reply:
yes, it is fixed now